File: frontend/src/types/match.ts
export interface Match {
  id: number
  user_id: number
  project_id: number
  swipe_id: number
  status: 'active' | 'completed' | 'closed'
  created_at: string
  closed_at?: string
  project?: MatchProject
  user?: MatchUser
}

export interface MatchProject {
  id: number
  title: string
  description: string
  owner: {
    id: number
    full_name: string
    avatar_file_id?: number
  }
}

export interface MatchUser {
  id: number
  full_name: string
  avatar_file_id?: number
  experience_years?: number
}

File: backend/app/schemas/match.py
from datetime import datetime
from typing import Optional
from app.models.match import MatchStatus
from pydantic import BaseModel

class MatchProjectOwnerResponse(BaseModel):
    id: int
    full_name: str
    avatar_file_id: Optional[int] = None
    
    model_config = {"from_attributes": True}


class MatchProjectResponse(BaseModel):
    id: int
    title: str
    description: str
    owner: MatchProjectOwnerResponse
    
    model_config = {"from_attributes": True}


class MatchUserResponse(BaseModel):
    id: int
    full_name: str
    avatar_file_id: Optional[int] = None
    experience_years: Optional[int] = None
    
    model_config = {"from_attributes": True}

class MatchResponse(BaseModel):
    id: int
    user_id: int
    project_id: int
    swipe_id: int
    status: MatchStatus
    created_at: Optional[datetime] = None
    closed_at: Optional[datetime] = None
    project: Optional[MatchProjectResponse] = None
    user: Optional[MatchUserResponse] = None
    
    model_config = {"from_attributes": True}
